doc/classes/SpringArm.xml

```diff
@@ -33,17 +41,25 @@
 			<argument index="0" name="RID" type="RID">
 			</argument>
 			<description>
+			Remove the given objects from the objects excluded from the collision check
 			</description>
 		</method>
 	</methods>
 	<members>
 		<member name="collision_mask" type="int" setter="set_collision_mask" getter="get_collision_mask" default="1">
+			The layers against which the collision check shall be done.
 		</member>
 		<member name="margin" type="float" setter="set_margin" getter="get_margin" default="0.01">
+			When the collision check is made, a candidate length for the SpringArm is given.
+			The margin is then subtracted to this length and the translation is applied to the child objects of the SpringArm.
+			This margin is useful for when the SpringArm has a Camera as a child node: without the margin, the Camera would be placed on the exact point of collision, while with the margin the Camera would be placed close to the point of collision.
 		</member>
 		<member name="shape" type="Shape" setter="set_shape" getter="get_shape">
+			The shape to use for the SpringArm.
+			When the Shape is set, the SpringArm will cast the shape on its z axis instead of a simple RayCast.
 		</member>
 		<member name="spring_length" type="float" setter="set_length" getter="get_length" default="1.0">
+			The maximum extent of the SpringArm. This is used as a length for both the RayCast and the ShapeCast done to calculate the desired position of the SpringArm's child nodes
```

Comment:
Lacks dot at end. There's also no ShapeCast class. If you mean that internal it casts a ray or a shape (but does not actually use a RayCast node for that), it would be better to spell out "both the ray and the shape casting".